### PR TITLE
Automation upgrade + node 16 

### PR DESCRIPTION
**Pull request by arun3528**
# COMPLETES #[SPARK-327612](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-327612)

## This pull request addresses

Upgrade the automation test framework , node version and fix linting 

## by making the following changes

< DESCRIBE YOUR CHANGES >

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [x] Tooling change
- [x] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -4,7 +4,6 @@ packages/node_modules/generator-ciscospark/generators/plugin/templates/src/_inde
 
 # The follwowing is copied from .gitignore
 .DS_Store
-/node_modules
```
Comment (mkesavan13) [inline .eslintignore:7]: Just curious to know why do we have to include node_modules for eslint

Comment (mkesavan13) [inline .eslintignore:7]: Just curious to know why do we have to include node_modules for eslint

File: docs/samples/browser-plugin-meetings/test/wdio/spec/mediatest.js [hunk 2]
```diff
@@ -1,41 +1,43 @@
 /* globals browser */
-import {expect} from 'chai';
+// import {expect} from 'chai';
```
Comment (mkesavan13) [inline docs/samples/browser-plugin-meetings/test/wdio/spec/mediatest.js:2]: If we are not using chai's expect, is it okay to remove the commented code as well?

File: package.json [hunk 3]
```diff
@@ -47,7 +47,7 @@
     "samples:build": "node ./tooling/index.js build --only-samples",
     "samples:serve": "webpack serve --color --env NODE_ENV=development",
     "samples:serve:integration": "webpack serve --color --env NODE_ENV=test",
-    "samples:test": "NODE_ENV=test wdio",
+    "samples:test": "NODE_ENV=test npx wdio run wdio.conf.js",
```
Comment (mkesavan13) [inline package.json:50]: I could see wdio available in devDependencies. Should we still be using npx here?

File: wdio.conf.js [hunk 4]
```diff
@@ -454,83 +525,62 @@ exports.config = {
...
    */
-  // afterSession: function (config, capabilities, specs) {
-  // },
   /**
    * Gets executed after all workers got shut down and the process is about to exit. It is not
    * possible to defer the end of the process using a promise.
    * @param {Object} exitCode 0 - success, 1 - fail
    */
-  // onComplete(exitCode) {
-  // }
```
Comment (mkesavan13) [inline wdio.conf.js:535]: If the commented hooks are removed, can we also remove their respective comments?

**Review by mkesavan13 — CHANGES_REQUESTED** [review]
Why have we removed tests for browser-multi-party-call? Aren't we using multi party calling anymore?

Just a question out of curiosity.

File: .eslintignore (hunk 1 above)
Comment (arun3528) [reply .eslintignore:7]: after eslint upgrade by default it was doing it so removed it , Did not throw any error to me

File: docs/samples/browser-plugin-meetings/test/wdio/spec/mediatest.js (hunk 2 above)
Comment (arun3528) [reply docs/samples/browser-plugin-meetings/test/wdio/spec/mediatest.js:2]: done removing

File: package.json (hunk 3 above)
Comment (arun3528) [reply package.json:50]: locally it was causing issue so added that

File: wdio.conf.js (hunk 4 above)
Comment (arun3528) [reply wdio.conf.js:535]: yep we could remove it for now

**Review by mkesavan13 — APPROVED** [review]
Looks good. Approving. Thanks for taking time to upgrade Arun. Probably a long pending debt. :-)

**Pull request merged**